include/engine/hdf5/HDF5WriterP.h

```diff
@@ -0,0 +1,224 @@
+/*
+ * HDF5WriterP.h
+ *
+ *  Created on: March 20, 2017
+ *      Author: Junmin
+ */
+
+#ifndef HDF5_WRITER_P_H_
+#define HDF5_WRITER_P_H_
+
+#include <iostream> //std::cout must be removed, only used for hello example
+#include <unistd.h> //sleep must be removed
+
+#include "core/Engine.h"
+#include "format/BP1Writer.h"
+
+// supported capsules
+#include "capsule/heap/STLVector.h"
+
+#include "ADIOS_MPI.h"
+
+#include "hdf5.h"
+
+namespace adios {
+
+typedef struct {
+  double _re; /*real part*/
+  double _im; /*imaginary part*/
+} ADIOS2_Complex_Double;
+
+typedef struct {
+  float _re; // real
+  float _im; // imaginary part
+} ADIOS2_Complex_Float;
+
+// using ADIOS2_Complex_Float = std::complex<float>;
+
+typedef struct {
+  long double _re; /*real part*/
+  long double _im; /*imaginary part*/
+} ADIOS2_Complex_LongDouble;
+
+class HDF5Writer : public Engine {
+
+public:
+  /**
+   * Constructor for single BP capsule engine, writes in BP format into a single
+   * heap capsule
+   * @param name unique name given to the engine
+   * @param accessMode
+   * @param mpiComm
+   * @param method
+   * @param debugMode
+   */
+
+  HDF5Writer(ADIOS &adios, const std::string name, const std::string accessMode,
+             MPI_Comm mpiComm, const Method &method,
+             const bool debugMode = false, const unsigned int cores = 1);
+
+  ~HDF5Writer();
```

Comment:
Destructor should be virtual and since the implementation is empty then just let the compiler create the default, i.:

virtual ~HDFWriter() = default;